src/FondOfSpryker/Zed/DiscountCustomMessages/Business/Model/DiscountCustomMessagesWriter.php

```diff
@@ -46,93 +46,43 @@ public function __construct(
      */
     public function updateDiscountCustomMessages(
         DiscountConfiguratorTransfer $discountConfiguratorTransfer
-    ): DiscountConfiguratorTransfer {
-        if ($discountConfiguratorTransfer->getDiscountCustomMessages()->count() === 0) {
-            return $this->createDiscountMessagesByLocale($discountConfiguratorTransfer);
-        }
-
+    ): DiscountConfiguratorTransfer {$locales = $this->localeFacade->getLocaleCollection();
         foreach ($discountConfiguratorTransfer->getDiscountCustomMessages() as $discountCustomMessageTransfer) {
-            $idDiscountCustomMessage = $discountCustomMessageTransfer->getIdDiscountCustomMessage();
+            if (!$discountCustomMessageTransfer->getIdDiscountCustomMessage()) {
+                $discountCustomMessageEntity = $this->hydrateDiscountCustomMessageEntity(
+                    $this->createDiscountCustomMessageEntity(),
+                    $discountCustomMessageTransfer,
+                    $discountConfiguratorTransfer->getDiscountGeneral()->getIdDiscount()
+                );
 
-            if ($idDiscountCustomMessage === null) {
-                $this->saveNewEntity($discountConfiguratorTransfer, $discountCustomMessageTransfer);
+                $discountCustomMessageEntity->save();
 
                 continue;
             }
 
-            $this->saveExistingEntity($discountConfiguratorTransfer, $discountCustomMessageTransfer);
-        }
+            $discountCustomMessageEntity = $this->customMessagesQueryContainer
+                ->queryDiscountCustomMessageByIdAndIdDiscount(
+                    $discountCustomMessageTransfer->getIdDiscountCustomMessage(),
+                    $discountCustomMessageTransfer->getIdDiscount()
+                );
 
-        return $discountConfiguratorTransfer;
-    }
+            if ($discountCustomMessageEntity === null) {
+                continue;
+            }
 
-    /**
-     * @param \Generated\Shared\Transfer\DiscountConfiguratorTransfer $discountConfiguratorTransfer
-     * @param \Generated\Shared\Transfer\DiscountCustomMessageTransfer $discountCustomMessageTransfer
-     *
-     * @return void
-     */
-    protected function saveExistingEntity(
-        DiscountConfiguratorTransfer $discountConfiguratorTransfer,
-        DiscountCustomMessageTransfer $discountCustomMessageTransfer
-    ): void {
-        $idDiscount = $discountConfiguratorTransfer->getDiscountGeneral()->getIdDiscount();
-        $idDiscountCustomMessage = $discountCustomMessageTransfer->getIdDiscountCustomMessage();
-
-        $discountCustomMessageEntity = $this->customMessagesQueryContainer
-            ->queryDiscountCustomMessageByIdAndIdDiscount($idDiscountCustomMessage, $idDiscount);
-
-        if (!$discountCustomMessageEntity) {
-            return;
+            $discountCustomMessageEntity->fromArray($discountCustomMessageTransfer->toArray());
+            $discountCustomMessageEntity->save();
```

Comment:
Same as above.. this logic should be in the persistence layer and use an EntityManager pattern.